flink-formats/flink-orc/src/main/java/org/apache/flink/orc/writer/HadoopNoCloseStream.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.orc.writer;
+
+import org.apache.hadoop.fs.FSDataOutputStream;
+import org.apache.hadoop.fs.FileSystem;
+
+import java.io.IOException;
+import java.io.OutputStream;
+
+/**
+ * This class is designed to not close the underlying flink stream to avoid exceptions when
+ * checkpointing.
+ */
+public class HadoopNoCloseStream extends FSDataOutputStream {
```

Comment:
This looks a bit scary, @gyfora?

Reply:
@liujiawinds could you please clarify why we need the `HadoopNoCloseStream` here?

Reply:
@morhidi Because `BulkWriter` needs to rely on a flink `FSDataOutputStream`, and ORC writer uses hadoop `FSDataOutputStream`. So I wrapped flink `FSDataOutputStream`.
     Additionally, BulkWriter closes the underlying ORC writer stream at checkpoint, which will cause flink to throw a `ClosedChannelException` if the close action is passed to flink `FSDataOutputStream`.

Reply:
qq: Is this covered in a test case somewhere?